app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,49 @@
+class SessionsController < ApplicationController
+  def create
+    auth_params ? oauth_create : default_create
+  end
+
+  def destroy
+    session[:user_id] = nil
+    redirect_to login_path
+  end
+
+  private
+
+  def auth_params
+    request.env['omniauth.auth']
+  end
+
+  def account_params
+    { uid: auth_params['uid'], provider: auth_params['provider'] }
+  end
+
+  def user_params
+    password = SecureRandom.hex(8)
+    user = User.find_or_create_by(email: auth_params['info']['email']) do |u|
+      u.password = password
```

Comment:
No, the code block executes only if user was created.
[http://apidock.com/rails/v4.2.7/ActiveRecord/Relation/find_or_create_by](url)